### PR TITLE
Nested edge integration with E2E pipelines

### PR DESCRIPTION
**Pull request by MaidaB**
- nested edge scripts added
- an optional nested edge job added to the pipelines
- DeployPLCs.ps1 needs to use azure cli methods because one of them is not supported by PowerShell cmdlets ( az container create  with a subnet parameter)

Orchestrated pipeline:
- New pipeline variable: NestedEdgeFlag (optional)

Standalone pipeline:
- New pipeline variables: EdgeVersion, NestedEdgeFlag (optional), NestedEdgeSshConnections (optional)
- IIoTEdgeConfig.cs is used to set the edge version, the nested edge flag and ssh connections array
- In the tests the two steps approach is used to copy pn.json file (first to the jumpbox VM and then to the machine in the lower nested edge layer)

### CODE REVIEW

**Review by cristipogacean — APPROVED** [review]

**Review by hansgschossmann — DISMISSED** [review]
:shipit:

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneTestTheory.cs [hunk 1]
```diff
@@ -47,11 +48,26 @@ public class A_PublishSingleNodeStandaloneTestTheory {
 
         [Fact, PriorityOrder(3)]
         public void Test_StartPublishingSingleNode_Expect_Success() {
-            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
-
+            var cts = new CancellationTokenSource(TimeSpan.FromMinutes(1));
```
Comment (mregen) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneTestTheory.cs:51]: >TimeSpan.FromMinutes(1) [](start = 50, length = 23)

why a magic number here when there used to be a configurable time?

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneTestTheory.cs [hunk 2]
```diff
@@ -66,9 +82,10 @@ public class A_PublishSingleNodeStandaloneTestTheory {
                 })
                 .ToArray();
 
+            cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
```
Comment (mregen) [inline e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneTestTheory.cs:85]: >TimeSpan.FromMinutes(5) [](start = 46, length = 23)

same as above

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneTestTheory.cs (hunk 1 above)
Comment (MaidaB) [reply e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneTestTheory.cs:51]: When running in the nested edge, this call is expected to time out, and waiting 30 minutes is too much, probably I should write it as a comment

File: e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneTestTheory.cs (hunk 2 above)
Comment (MaidaB) [reply e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneTestTheory.cs:85]: this one can be the default one, I will change it

**Review by cristipogacean — APPROVED** [review]

**Pull request merged**